patches/1.15.8.1/nginx-1.15.8-upstream_client_certificate.patch

```diff
@@ -0,0 +1,25 @@
+diff --git a/nginx-1.15.8/src/http/ngx_http_upstream.c b/nginx-1.15.8/src/http/ngx_http_upstream.c
+index 90710557..539a4db9 100644
+--- a/nginx-1.15.8/src/http/ngx_http_upstream.c
++++ b/nginx-1.15.8/src/http/ngx_http_upstream.c
+@@ -8,6 +8,9 @@
+ #include <ngx_config.h>
+ #include <ngx_core.h>
+ #include <ngx_http.h>
++#if (NGX_HTTP_LUA_KONG)
++#include <ngx_http_lua_kong_module.h>
++#endif
+ 
+ 
+ #if (NGX_HTTP_CACHE)
+@@ -1707,6 +1710,10 @@ ngx_http_upstream_ssl_init_connection(ngx_http_request_t *r,
+         }
+     }
+ 
++#if (NGX_HTTP_LUA_KONG)
++    ngx_http_lua_kong_set_upstream_ssl(r, c);
++#endif
++
+     r->connection->log->action = "SSL handshaking to upstream";
+ 
+     rc = ngx_ssl_handshake(c);
```

Comment:
I believe that this patch should belong to our openresty-patches repository? It's not going to be a good thing if we split them between openresty-patches and this repository (also consider we have a few non Kong-related ones in openresty-build-tools as well).

Also, this patch should, if possible, set `HAVE_KONG_SET_UPSTREAM_SSL` (or similar), which, if not set, would cause `ngx_http_lua_kong_ffi_set_upstream_client_cert_and_key()` to return `NGX_DECLINED` (or similar), which, in turn, would cause `kong.tls.set_upstream_cert_and_key()` to return an error basically saying the feature is "unsupported because of missing patch xyz".

Reply:
Okay, I will move the patch to the `openresty-patches` repo.

For the macro guard, I discussed with @thibaultcha and here is what we will do:

https://trello.com/c/V88fMBlQ/1883-add-patch-and-nginx-module-checks-for-kong-core

It falls outside of this PR's scope though. So let's merge this PR first and open another PR to do that.